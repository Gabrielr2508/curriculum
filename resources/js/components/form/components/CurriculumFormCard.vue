<template>
	<v-flex xs12 sm6 offset-sm3>
		<v-card>
			<v-card-title primary-title>
			</v-card-title>
			<v-card-text>
				<v-form>
					<v-container>
						<v-layout column>
							<v-flex
								xs12
							>
								<v-text-field
									label="Nome*"
									outline
									prepend-icon="person"
								></v-text-field>
							</v-flex>
							<v-flex
								xs12
							>
								<v-text-field
									label="E-mail*"
									outline
									prepend-icon="email"
								></v-text-field>
							</v-flex>
							<v-flex
								xs12
							>
								<v-text-field
									label="Telefone*"
									outline
									prepend-icon="call"
								></v-text-field>
							</v-flex>
							<v-flex
								xs12
							>
								<v-text-field
									label="Linkedin"
									outline
								>
									<template v-slot:prepend>
											<div class="v-input__icon v-input__icon--prepend">
												<i aria-hidden="true" class="v-icon material-icons icon-linkedin1 theme--light"></i>
											</div>
									</template>
								</v-text-field>
							</v-flex>
							<v-flex
								xs12
							>
								<v-text-field
									label="GitHub"
									outline
								>
									<template v-slot:prepend>
											<div class="v-input__icon v-input__icon--prepend">
												<i aria-hidden="true" class="v-icon material-icons icon-github theme--light"></i>
											</div>
									</template>
								</v-text-field>
							</v-flex>
							<v-flex
								xs12
							>
								<v-text-field
									v-model="filename"
									@click.native="onFocus"
									label="Currículo (pdf)*"
									outline
									prepend-icon="attach_file"
									ref="fileTextField"
								/>
								<input
									:multiple="false"
									:accept="accept"
									@change="onFileChange"
									ref="fileInput"
									class="d-none"
									type="file"
								>
							</v-flex>
							<v-flex xs12>
								<v-select
									v-model="areas"
									:items="options"
									label="Áreas de interesse"
									multiple
									chips
									outline
									deletable-chips
									prepend-icon="work"
									small-chips
								/>
							</v-flex>
							<v-flex
								xs12
							>
								<v-switch
									label="Estágio"
									color="primary"
								/>
							</v-flex>
							<v-flex xs12>
								<v-subheader class="red--text">* campos obrigatórios</v-subheader>
							</v-flex>
						</v-layout>
					</v-container>
				</v-form>
			</v-card-text>
			<v-card-actions>
				<v-btn
					absolute
					dark
					fab
					bottom
					right
					color="primary"
					>
					<v-icon>send</v-icon>
				</v-btn>
			</v-card-actions>
		</v-card>
	</v-flex>
</template>

<script>
export default {
	data() {
		return {
			card_text: 'Test',
			options: [
				'Front-end',
				'Back-end',
				'Administrativo',
				'Financeiro',
				'Cozinha',
				'Limpeza',
				'DevOps',
				'Design',
				'UI/UX'
			],
			areas: [],
			filename: '',
		}
	},

	methods: {
		getFormData(files){
			const data = new FormData();
			[...files].forEach(file => {
				data.append('data', file, file.name); // currently only one file at a time
			});
			return data;
		},

		onFocus(){
			if (!this.disabled) {
				this.$refs.fileInput.click();
			}
		},

		onFileChange($event){
			const files = $event.target.files || $event.dataTransfer.files;
			const form = this.getFormData(files);

			if (files) {
				if (files.length > 0) {
					this.filename = [...files].map(file => file.name).join(', ');
				} else {
					this.filename = null;
				}
			} else {
				this.filename = $event.target.value.split('\\').pop();
			}

			this.$emit('input', this.filename);
			this.$emit('formData', form);
		}
	}
}
</script>

<style>

</style>
