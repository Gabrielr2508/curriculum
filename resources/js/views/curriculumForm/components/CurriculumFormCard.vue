<template>
	<v-flex xs12 sm6 offset-sm3>
		<v-card>
			<v-img
				src="../img/card-bg.jpg"
				aspect-ratio="2.75"
				class="white--text"
			>
				<v-container fill-height fluid>
					<v-layout fill-height align-end>
						<v-flex xs12>
							<span class="headline">May your force be with us</span>
						</v-flex>
					</v-layout>
				</v-container>
			</v-img>

			<v-card-text>
				<v-form
					v-model="valid"
					ref="curriculumForm"
				>
					<v-container>
						<v-layout column>
							<v-flex
								xs12
							>
								<v-text-field
									v-model="formData.name"
									:rules="rules.name"
									label="Nome*"
									outline
									prepend-icon="person"
									required
								></v-text-field>
							</v-flex>
							<v-flex
								xs12
							>
								<v-text-field
									v-model="formData.email"
									:rules="rules.email"
									label="E-mail*"
									outline
									prepend-icon="email"
									required
								></v-text-field>
							</v-flex>
							<v-flex
								xs12
							>
								<v-text-field
									v-model="formData.phone"
									:rules="rules.phone"
									mask="(##) ##### - ####"
									label="Celular*"
									outline
									required
								>
									<template v-slot:prepend>
											<div class="v-input__icon v-input__icon--prepend">
												<i aria-hidden="true" class="v-icon material-icons icon-whatsapp theme--light"></i>
											</div>
									</template>
								</v-text-field>
							</v-flex>
							<v-flex
								xs12
							>
								<v-text-field
									label="Linkedin"
									v-model="formData.linkedin"
									:rules="rules.linkedin"
									outline
								>
									<template v-slot:prepend>
											<div class="v-input__icon v-input__icon--prepend">
												<i aria-hidden="true" class="v-icon material-icons icon-linkedin1 theme--light"></i>
											</div>
									</template>
								</v-text-field>
							</v-flex>
							<v-flex
								xs12
							>
								<v-text-field
									label="GitHub"
									v-model="formData.github"
									:rules="rules.github"
									outline
								>
									<template v-slot:prepend>
											<div class="v-input__icon v-input__icon--prepend">
												<i aria-hidden="true" class="v-icon material-icons icon-github theme--light"></i>
											</div>
									</template>
								</v-text-field>
							</v-flex>
							<v-flex
								xs12
							>
								<v-text-field
									v-model="formData.filename"
									@click.native="onFocus"
									label="Currículo (pdf)*"
									outline
									prepend-icon="attach_file"
									ref="fileTextField"
									required
									:rules="rules.filename"
								/>
								<input
									:multiple="false"
									accept="application/pdf"
									@change="onFileChange"
									ref="fileInput"
									class="d-none"
									type="file"
								>
							</v-flex>
							<v-flex xs12>
								<v-select
									v-model="formData.areas"
									:items="options"
									:rules="rules.areas"
									label="Áreas de interesse*"
									multiple
									chips
									outline
									deletable-chips
									prepend-icon="work"
									small-chips
									required
								/>
							</v-flex>
							<v-flex
								xs12
							>
								<v-switch
									label="Estágio"
									v-model="formData.is_internship"
									color="primary"
								/>
							</v-flex>
							<v-flex xs12>
								<span class="red--text caption">* campos obrigatórios</span>
							</v-flex>
							<vue-recaptcha
								:sitekey="sitekey"
								ref="invisibleRecaptcha"
								size="invisible"
								badge="bottomleft"
								@verify="onVerifyRecaptcha"
								@expired="resetRecaptcha"
							/>
						</v-layout>
					</v-container>
				</v-form>
			</v-card-text>
			<v-card-actions>
				<v-tooltip top>
					<template v-slot:activator="{ on }">
						<v-btn
							v-on="on"
							@click="handleSubmit"
							absolute
							dark
							fab
							bottom
							right
							color="primary"
						>
							<v-progress-circular
								v-if="disableSubmit"
								indeterminate
							/>
							<v-icon v-else>send</v-icon>
						</v-btn>
					</template>
					<span>Enviar</span>
				</v-tooltip>
			</v-card-actions>
		</v-card>
	</v-flex>
</template>

<script>
import env from '../../../env';
import VueRecaptcha from 'vue-recaptcha';
import validators from '../../../common/validators';

export default {
	components: {
		VueRecaptcha,
	},
	data() {
		return {
			sitekey: env.SITE_KEY,
			options: [
				'Front-end',
				'Back-end',
				'Administrativo',
				'Financeiro',
				'Cozinha',
				'Limpeza',
				'DevOps',
				'Design',
				'UI/UX'
			],
			valid: true,
			disableSubmit: false,
			formData: {
				areas: null,
				filename: '',
				name: '',
				email: '',
				phone: '',
				linkedin: '',
				github: '',
				is_internship: false,
				file: null,
				recaptcha_response: null,
			},
			rules: {
				...validators,
			}
		}
	},

	methods: {
		getFormData(pdf){
			const data = new FormData();
			Object.keys(this.formData)
				.forEach(key => {
					data.append(key, this.formData[key]);
				});

			return data;
		},

		onFocus(){
			if (!this.disabled) {
				this.$refs.fileInput.click();
			}
		},

		onFileChange(){
			const pdf = this.$refs.fileInput.files[0];
			if (pdf) {
				this.formData.filename = pdf.name;
				this.formData.file = pdf;
			} else {
				this.formData.filename = null;
				this.formData.file = null;
			}
		},

		onVerifyRecaptcha(response) {
			this.formData.recaptcha_response = response;

			this.resetRecaptcha();

			console.log('Verify: ' + response)

			axios.post(
				'/curriculum',
				this.getFormData(),
				{
					headers: {
						'Content-Type': 'multipart/form-data'
					}
				}
			)
			.then((response) => {
				console.log(response);
			})
			.catch((error) => {
				console.log(error.response);
			})
			.then(() => {
				this.disableSubmit = false;
			});
		},

		resetRecaptcha() {
			this.$refs.invisibleRecaptcha.reset();
		},

		handleSubmit() {
			if (!this.disableSubmit && this.$refs.curriculumForm.validate()) {
				this.disableSubmit = true;
				this.$refs.invisibleRecaptcha.execute();
			}
		}
	}
}
</script>

<style>

</style>
