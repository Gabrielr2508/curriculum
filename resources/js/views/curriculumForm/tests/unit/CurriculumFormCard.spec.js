import { shallowMount } from '@vue/test-utils';
import Vue from 'vue';
import Vuetify from 'vuetify';

import CurriculumFormCard from '@/views/curriculumForm/components/CurriculumFormCard';
import VueRecaptcha from 'vue-recaptcha';

Vue.use(Vuetify);

describe('CurriculumFormCard', () => {
    const build = () => {
        const wrapper = shallowMount(CurriculumFormCard);

        return {
            wrapper,
            vueRecaptcha: () => wrapper.find(VueRecaptcha),
        }
    }

    it('renders the component', () => {
        // arrange
        const { wrapper } = build();

        // assert
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('renders main child components', () => {
        // arrange
        const { vueRecaptcha } = build();

        // assert
        expect(vueRecaptcha().exists()).toBe(true);
    });

    it('passes a binded sitekey prop to vue-recaptcha component', () => {
        // arrange
        const { wrapper, vueRecaptcha } = build();

        wrapper.setData({
            sitekey: 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaa',
        });

        // assert
        expect(vueRecaptcha().vm.sitekey).toBe(wrapper.vm.sitekey);
    });
});
