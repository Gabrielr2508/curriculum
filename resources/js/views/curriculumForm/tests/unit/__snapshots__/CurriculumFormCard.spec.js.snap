// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurriculumFormCard renders the component 1`] = `
<v-flex-stub tag="div" xs12="" sm6="" offset-sm3="">
  <v-card-stub tag="div">
    <v-img-stub aspectratio="2.75" src="../img/card-bg.jpg" position="center center" transition="fade-transition" class="white--text">
      <v-container-stub tag="div" fill-height="" fluid="">
        <v-layout-stub tag="div" fill-height="" align-end="">
          <v-flex-stub tag="div" xs12=""><span class="headline font-weight-light">May your force be with us</span></v-flex-stub>
        </v-layout-stub>
      </v-container-stub>
    </v-img-stub>
    <v-card-text-stub>
      <v-form-stub value="true">
        <v-container-stub tag="div">
          <v-layout-stub tag="div" column="">
            <v-flex-stub tag="div" xs12="">
              <v-text-field-stub rules="v => !!v || 'Esse campo é obrigatório',v => v &amp;&amp; v.length >= 10 || 'O nome precisa ter pelo menos 10 caracteres'" label="Nome*" outline="" prepend-icon="person" required="" value=""></v-text-field-stub>
            </v-flex-stub>
            <v-flex-stub tag="div" xs12="">
              <v-text-field-stub rules="v => !!v || 'Esse campo é obrigatório',v => {
    const pattern = /^(([^<>()[\\]\\\\.,;:\\s@&quot;]+(\\.[^<>()[\\]\\\\.,;:\\s@&quot;]+)*)|(&quot;.+&quot;))@((\\[[0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3}])|(([a-zA-Z\\-0-9]+\\.)+[a-zA-Z]{2,}))$/;
    return pattern.test(v) || 'O E-mail precisa ser válido';
  }" label="E-mail*" outline="" prepend-icon="email" required="" value=""></v-text-field-stub>
            </v-flex-stub>
            <v-flex-stub tag="div" xs12="">
              <v-text-field-stub rules="v => !!v || 'Esse campo é obrigatório',v => {
    const pattern = /^(1[2-9]{1})|([2-9]{1}\\d{1})(9\\d{8})$/;
    return pattern.test(v) || 'O número de celular precisa ser válido';
  }" mask="(##) ##### - ####" label="Celular*" outline="" required="" value=""></v-text-field-stub>
            </v-flex-stub>
            <v-flex-stub tag="div" xs12="">
              <v-text-field-stub label="Linkedin" rules="v => {
    if (v.length === 0) {
      return true;
    }

    const pattern = /^http(s)?:\\/\\/([\\w]+\\.)?linkedin\\.com\\/in\\/[A-z0-9_-]+\\/?$/;
    return pattern.test(v) || 'O link do LinkedIn precisa ser válido';
  }" outline="" value=""></v-text-field-stub>
            </v-flex-stub>
            <v-flex-stub tag="div" xs12="">
              <v-text-field-stub label="GitHub" rules="v => {
    if (v.length === 0) {
      return true;
    }

    const pattern = /^http(s)?:\\/\\/(www\\.)?github\\.com\\/[A-z0-9_-]+\\/?$/;
    return pattern.test(v) || 'O link do GitHub precisa ser válido';
  }" outline="" value=""></v-text-field-stub>
            </v-flex-stub>
            <v-flex-stub tag="div" xs12="">
              <v-text-field-stub label="Currículo (pdf)*" outline="" prepend-icon="attach_file" required="" rules="v => !!v || 'Esse campo é obrigatório'" value=""></v-text-field-stub> <input accept="application/pdf" type="file" class="d-none">
            </v-flex-stub>
            <v-flex-stub tag="div" xs12="">
              <v-select-stub multiple="true" items="Front-end,Back-end,Administrativo,Financeiro,Cozinha,Limpeza,DevOps,Design,UI/UX" rules="v => !!v &amp;&amp; v.length !== 0 || 'Escolha pelo menos uma área'" label="Áreas de interesse*" chips="" outline="" deletable-chips="" prepend-icon="work" small-chips="" required=""></v-select-stub>
            </v-flex-stub>
            <v-flex-stub tag="div" xs12="">
              <v-switch-stub color="primary" errorcount="1" errormessages="" messages="" rules="" successmessages="" backgroundcolor="" label="Estágio" ripple="true" valuecomparator="function deepEqual(a, b) {
    if (a === b) return true;
    if (a instanceof Date &amp;&amp; b instanceof Date) {
        // If the values are Date, they were convert to timestamp with getTime and compare it
        if (a.getTime() !== b.getTime()) return false;
    }
    if (a !== Object(a) || b !== Object(b)) {
        // If the values aren't objects, they were already checked for equality
        return false;
    }
    var props = Object.keys(a);
    if (props.length !== Object.keys(b).length) {
        // Different number of props, don't bother to check
        return false;
    }
    return props.every(function (p) {
        return deepEqual(a[p], b[p]);
    });
}"></v-switch-stub>
            </v-flex-stub>
            <v-flex-stub tag="div" xs12=""><span class="red--text caption">* campos obrigatórios</span></v-flex-stub>
            <vue-recaptcha-stub sitekey="6LfuXagUAAAAAHLGJo4BaJKb9lJ_hiBsYmrvXVHC" badge="bottomleft" size="invisible"></vue-recaptcha-stub>
          </v-layout-stub>
        </v-container-stub>
      </v-form-stub>
    </v-card-text-stub>
    <v-card-actions-stub>
      <v-tooltip-stub opendelay="200" closedelay="200" contentclass="" fixed="true" top="true" maxwidth="auto" nudgebottom="0" nudgeleft="0" nudgeright="0" nudgetop="0" nudgewidth="0" debounce="0" tag="span"> <span>Enviar</span></v-tooltip-stub>
    </v-card-actions-stub>
  </v-card-stub>
</v-flex-stub>
`;
